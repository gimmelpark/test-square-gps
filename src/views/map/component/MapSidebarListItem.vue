<template>
  <VListItem>
    <VListItemContent>
      <VListItemTitle class="d-flex align-center">
        <VBtn
          x-small
          icon
          class="mr-3 my-5"
          :disabled="isLoading"
          @click.stop="onEditButtonClick"
        >
          <VIcon>{{ isEditing ? "mdi-content-save" : "mdi-pencil" }}</VIcon>
        </VBtn>

        <div v-if="!isEditing">
          {{ point.title }}
        </div>

        <VTextField v-else v-model="editedTitle" dense @click.native.stop />
      </VListItemTitle>

      <VListItemSubtitle class="text-wrap">
        {{ point.address }}
      </VListItemSubtitle>
    </VListItemContent>

    <VListItemAction>
      <VBtn
        :disabled="isLoading"
        icon
        color="red"
        @click.stop="$emit('delete')"
      >
        <VIcon>mdi-close</VIcon>
      </VBtn>
    </VListItemAction>
  </VListItem>
</template>

<script>
export default {
  props: {
    point: {
      type: Object,
      required: true,
    },
    isLoading: Boolean,
  },

  data: () => ({
    isEditing: false,
    editedTitle: "",
  }),

  methods: {
    onEditButtonClick() {
      if (this.isEditing) {
        this.$emit("save-changes", this.editedTitle);
      } else {
        this.editedTitle = this.point.title;
      }

      this.isEditing = !this.isEditing;
    },
  },
};
</script>
