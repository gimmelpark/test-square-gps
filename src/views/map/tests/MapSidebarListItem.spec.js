import { createLocalVue, shallowMount } from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'

import MapSidebarListItem from "../component/MapSidebarListItem.vue";



describe('MapSidebarListItem', () => {
  let wrapper;

  const defaultProps = {
    point: {
      id: 1,
      title: 'asd',
      address: 'qwe'
    },
    isLoading: false
  }

  beforeEach(() => {
    Vue.use(Vuetify)
    const localVue = createLocalVue()
    let vuetify

    vuetify = new Vuetify()

    wrapper = shallowMount(MapSidebarListItem, {
      localVue,
      vuetify,
      propsData: defaultProps
    })
  })

  afterEach(() => {
    wrapper.destroy()
  })

  describe('snapshots', () => {
    it('snapshot with default props', () => {
      expect(wrapper.element).toMatchSnapshot()
    })

    it('snapshot with isLoading: true', async () => {
      await wrapper.setProps({
        isLoading: true,
      })

      expect(wrapper.element).toMatchSnapshot()
    })

    it('snapshot in editing mode', async () => {
      await wrapper.setData({
        isEditing: true
      })

      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('methods', () => {
    describe('onEditButtonClick', () => {
      it("should call $emit with passed arguments and set isEditing: false if isEditing is true", async () => {
        await wrapper.setData({
          editedTitle: 'aaaa',
          isEditing: true
        })

        wrapper.vm.onEditButtonClick()

        expect(wrapper.emitted()['save-changes'][0]).toEqual(['aaaa'])
        expect(wrapper.vm.isEditing).toBe(false)
      })

      it("should set editedTitle and set isEditing: true if isEditing is false", async () => {

        wrapper.vm.onEditButtonClick()

        expect(wrapper.vm.editedTitle).toBe('asd')
        expect(wrapper.vm.isEditing).toBe(false)
      })
    })
  })
})