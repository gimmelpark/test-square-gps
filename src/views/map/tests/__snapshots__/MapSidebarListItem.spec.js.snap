// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapSidebarListItem common should match snapshot in editing mode 1`] = `
<vlistitem-stub
  activeclass=""
  tag="div"
>
  <vlistitemcontent-stub
    tag="div"
  >
    <vlistitemtitle-stub
      class="d-flex align-center"
      tag="div"
    >
      <vbtn-stub
        activeclass=""
        class="mr-3 my-5"
        icon="true"
        tag="button"
        type="button"
        xsmall="true"
      >
        <vicon-stub>
          mdi-content-save
        </vicon-stub>
      </vbtn-stub>
       
      <vtextfield-stub
        backgroundcolor=""
        clearicon="$clear"
        dense="true"
        errorcount="1"
        errormessages=""
        loaderheight="2"
        messages=""
        rules=""
        successmessages=""
        type="text"
        value=""
      />
    </vlistitemtitle-stub>
     
    <vlistitemsubtitle-stub
      class="text-wrap"
      tag="div"
    >
      
      qwe
    
    </vlistitemsubtitle-stub>
  </vlistitemcontent-stub>
   
  <vlistitemaction-stub>
    <vbtn-stub
      activeclass=""
      color="red"
      icon="true"
      tag="button"
      type="button"
    >
      <vicon-stub>
        mdi-close
      </vicon-stub>
    </vbtn-stub>
  </vlistitemaction-stub>
</vlistitem-stub>
`;

exports[`MapSidebarListItem common should match snapshots with default props 1`] = `
<vlistitem-stub
  activeclass=""
  tag="div"
>
  <vlistitemcontent-stub
    tag="div"
  >
    <vlistitemtitle-stub
      class="d-flex align-center"
      tag="div"
    >
      <vbtn-stub
        activeclass=""
        class="mr-3 my-5"
        icon="true"
        tag="button"
        type="button"
        xsmall="true"
      >
        <vicon-stub>
          mdi-pencil
        </vicon-stub>
      </vbtn-stub>
       
      <div>
        
        asd
      
      </div>
    </vlistitemtitle-stub>
     
    <vlistitemsubtitle-stub
      class="text-wrap"
      tag="div"
    >
      
      qwe
    
    </vlistitemsubtitle-stub>
  </vlistitemcontent-stub>
   
  <vlistitemaction-stub>
    <vbtn-stub
      activeclass=""
      color="red"
      icon="true"
      tag="button"
      type="button"
    >
      <vicon-stub>
        mdi-close
      </vicon-stub>
    </vbtn-stub>
  </vlistitemaction-stub>
</vlistitem-stub>
`;

exports[`MapSidebarListItem common should match snapshots with isLoading: true 1`] = `
<vlistitem-stub
  activeclass=""
  tag="div"
>
  <vlistitemcontent-stub
    tag="div"
  >
    <vlistitemtitle-stub
      class="d-flex align-center"
      tag="div"
    >
      <vbtn-stub
        activeclass=""
        class="mr-3 my-5"
        disabled="true"
        icon="true"
        tag="button"
        type="button"
        xsmall="true"
      >
        <vicon-stub>
          mdi-pencil
        </vicon-stub>
      </vbtn-stub>
       
      <div>
        
        asd
      
      </div>
    </vlistitemtitle-stub>
     
    <vlistitemsubtitle-stub
      class="text-wrap"
      tag="div"
    >
      
      qwe
    
    </vlistitemsubtitle-stub>
  </vlistitemcontent-stub>
   
  <vlistitemaction-stub>
    <vbtn-stub
      activeclass=""
      color="red"
      disabled="true"
      icon="true"
      tag="button"
      type="button"
    >
      <vicon-stub>
        mdi-close
      </vicon-stub>
    </vbtn-stub>
  </vlistitemaction-stub>
</vlistitem-stub>
`;

exports[`MapSidebarListItem common snapshots with default props 1`] = `
<vlistitem-stub
  activeclass=""
  tag="div"
>
  <vlistitemcontent-stub
    tag="div"
  >
    <vlistitemtitle-stub
      class="d-flex align-center"
      tag="div"
    >
      <vbtn-stub
        activeclass=""
        class="mr-3 my-5"
        icon="true"
        tag="button"
        type="button"
        xsmall="true"
      >
        <vicon-stub>
          mdi-pencil
        </vicon-stub>
      </vbtn-stub>
       
      <div>
        
        asd
      
      </div>
    </vlistitemtitle-stub>
     
    <vlistitemsubtitle-stub
      class="text-wrap"
      tag="div"
    >
      
      qwe
    
    </vlistitemsubtitle-stub>
  </vlistitemcontent-stub>
   
  <vlistitemaction-stub>
    <vbtn-stub
      activeclass=""
      color="red"
      icon="true"
      tag="button"
      type="button"
    >
      <vicon-stub>
        mdi-close
      </vicon-stub>
    </vbtn-stub>
  </vlistitemaction-stub>
</vlistitem-stub>
`;

exports[`MapSidebarListItem snapshots snapshot in editing mode 1`] = `
<vlistitem-stub
  activeclass=""
  tag="div"
>
  <vlistitemcontent-stub
    tag="div"
  >
    <vlistitemtitle-stub
      class="d-flex align-center"
      tag="div"
    >
      <vbtn-stub
        activeclass=""
        class="mr-3 my-5"
        icon="true"
        tag="button"
        type="button"
        xsmall="true"
      >
        <vicon-stub>
          mdi-content-save
        </vicon-stub>
      </vbtn-stub>
       
      <vtextfield-stub
        backgroundcolor=""
        clearicon="$clear"
        dense="true"
        errorcount="1"
        errormessages=""
        loaderheight="2"
        messages=""
        rules=""
        successmessages=""
        type="text"
        value=""
      />
    </vlistitemtitle-stub>
     
    <vlistitemsubtitle-stub
      class="text-wrap"
      tag="div"
    >
      
      qwe
    
    </vlistitemsubtitle-stub>
  </vlistitemcontent-stub>
   
  <vlistitemaction-stub>
    <vbtn-stub
      activeclass=""
      color="red"
      icon="true"
      tag="button"
      type="button"
    >
      <vicon-stub>
        mdi-close
      </vicon-stub>
    </vbtn-stub>
  </vlistitemaction-stub>
</vlistitem-stub>
`;

exports[`MapSidebarListItem snapshots snapshot with default props 1`] = `
<vlistitem-stub
  activeclass=""
  tag="div"
>
  <vlistitemcontent-stub
    tag="div"
  >
    <vlistitemtitle-stub
      class="d-flex align-center"
      tag="div"
    >
      <vbtn-stub
        activeclass=""
        class="mr-3 my-5"
        icon="true"
        tag="button"
        type="button"
        xsmall="true"
      >
        <vicon-stub>
          mdi-pencil
        </vicon-stub>
      </vbtn-stub>
       
      <div>
        
        asd
      
      </div>
    </vlistitemtitle-stub>
     
    <vlistitemsubtitle-stub
      class="text-wrap"
      tag="div"
    >
      
      qwe
    
    </vlistitemsubtitle-stub>
  </vlistitemcontent-stub>
   
  <vlistitemaction-stub>
    <vbtn-stub
      activeclass=""
      color="red"
      icon="true"
      tag="button"
      type="button"
    >
      <vicon-stub>
        mdi-close
      </vicon-stub>
    </vbtn-stub>
  </vlistitemaction-stub>
</vlistitem-stub>
`;

exports[`MapSidebarListItem snapshots snapshot with isLoading: true 1`] = `
<vlistitem-stub
  activeclass=""
  tag="div"
>
  <vlistitemcontent-stub
    tag="div"
  >
    <vlistitemtitle-stub
      class="d-flex align-center"
      tag="div"
    >
      <vbtn-stub
        activeclass=""
        class="mr-3 my-5"
        disabled="true"
        icon="true"
        tag="button"
        type="button"
        xsmall="true"
      >
        <vicon-stub>
          mdi-pencil
        </vicon-stub>
      </vbtn-stub>
       
      <div>
        
        asd
      
      </div>
    </vlistitemtitle-stub>
     
    <vlistitemsubtitle-stub
      class="text-wrap"
      tag="div"
    >
      
      qwe
    
    </vlistitemsubtitle-stub>
  </vlistitemcontent-stub>
   
  <vlistitemaction-stub>
    <vbtn-stub
      activeclass=""
      color="red"
      disabled="true"
      icon="true"
      tag="button"
      type="button"
    >
      <vicon-stub>
        mdi-close
      </vicon-stub>
    </vbtn-stub>
  </vlistitemaction-stub>
</vlistitem-stub>
`;
